lib/csv_sniffer.ex

```diff
@@ -64,24 +64,64 @@ defmodule CsvSniffer do
     ~r'(?P<delim>#{@delimiters_regex})(?P<quote>["\']).*?(?P=quote)(?P=delim)'sm,
     #  ".*?",
     ~r'(?:^|\n)(?P<quote>["\']).*?(?P=quote)(?P<delim>#{@delimiters_regex})'sm,
-    ~r'(?:^|\r)(?P<quote>["\']).*?(?P=quote)(?P<delim>#{@delimiters_regex})'sm,
     # ,".*?"
     ~r'(?P<delim>#{@delimiters_regex})(?P<quote>["\']).*?(?P=quote)(?:$|\n)'sm,
-    ~r'(?P<delim>#{@delimiters_regex})(?P<quote>["\']).*?(?P=quote)(?:$|\r)'sm,
     #  ".*?" (no delim)
-    ~r'(?:^|\n)(?P<quote>["\']).*?(?P=quote)(?:$|\n)'sm,
-    ~r'(?:^|\r)(?P<quote>["\']).*?(?P=quote)(?:$|\r)'sm
+    ~r'(?:^|\n)(?P<quote>["\']).*?(?P=quote)(?:$|\n)'sm
   ]
 
   defp run_quote_regex(sample) do
     Enum.find_value(@quote_regex, {[], []}, fn regex ->
-      case Regex.scan(regex, sample, capture: :all_names) do
-        [] -> false
-        matches -> {Regex.names(regex), matches}
+      names = Regex.names(regex)
+
+      with matches <- Regex.scan(regex, sample, capture: :all_names),
+           [_ | _] = matches <- Enum.filter(matches, &header_compatible?({names, &1}, sample)) do
+        {names, matches}
+      else
+        _ -> false
```

Comment:
now we discard any `matches` that is not "header compatible", that is the delimiter character defined is not the delimiter character candidate by inspecting only the header line